Realm/RLMSyncPermissionValue.mm

```diff
@@ -76,6 +78,27 @@ - (instancetype)initWithRealmPath:(NSString *)path
         _accessLevel = accessLevel;
         _path = path;
         _userID = userID;
+        if (!_userID) {
+            @throw RLMException(@"A permission value cannot be created without a valid user ID");
+        }
+        _updatedAt = [NSDate date];
+    }
+    return self;
+}
+
+- (instancetype)initWithRealmPath:(NSString *)path
```

Comment:
Yeah, the effective method name is different (`initWithRealmPath:user:...`) versus (`initWithRealmPath:email`).